### ISSUE TITLE
Plugin that assesses the risk of a file not being well characterised 

### ISSUE THREAD

**Issue by jmaferreira** [keeps/roda#642]
The plugin should detect the files that don't have a format identified and associate them to a risk. Then we should be able to run siegfried on those files to remove the risk.
- The plugin should operate over IndexedFiles
- The plugin should have 3 parameters flags (mime type, promo, format)
- The plugin should check for the existence of a specific Risk (if it doesn't exist, create it - see how "AIP corruption risk assessment" works).
- Affected files should be added as incidences to the created risk
- Make sure the plugin is well described
- Generate PREMIS event
- Put on the plugin outcome details the summary of checked files

**Comment by rui-castro** [comment]
@luis100 @hsilva-keep done in PR #702 

NOTE: the risk information (description and notes) needs to be improved. I don't know how...
* https://github.com/rui-castro/roda/blob/rcastro_642_missing_format_plugin/roda-core/roda-plugins/misc/roda-plugin-misc-missingformat/src/main/resources/default/data/storage/risk/urn:missingformat:r1.json
* https://github.com/rui-castro/roda/blob/rcastro_642_missing_format_plugin/roda-core/roda-plugins/misc/roda-plugin-misc-missingformat/src/main/resources/default/data/storage/risk/urn:missingmimetype:r1.json
* https://github.com/rui-castro/roda/blob/rcastro_642_missing_format_plugin/roda-core/roda-plugins/misc/roda-plugin-misc-missingformat/src/main/resources/default/data/storage/risk/urn:missingpronom:r1.json

**Comment by jmaferreira** [comment]
We can fix the descriptions. No problem.
I'm just wondering if the urn should have the exact name of the plugin in its ID for better tracking. What do you think @luis100

**Comment by rui-castro** [comment]
plugin class `org.roda.core.plugins.riskManagement.FileNotCharacterizedRiskAssessmentPlugin`

**Comment by rui-castro** [comment]
@luis100 @hsilva-keep @jmaferreira done in PR #715

**Comment by luis100** [comment]
- [ ] Output is unclear, does 1 means "yes" and 0 means "no"?

```
File is not comprehensively characterized.
Missing format designation: 1
Missing mimetype: 0
Missing PRONOM UID: 0
```

- [ ] Plugin created incidence when no incidence seems to be needed

Tried the plugin and it stated is was missing format format designation but it wasn't.

![report](https://cloud.githubusercontent.com/assets/631728/20680800/7eac436a-b598-11e6-840f-c3e8fd68bce1.png)
![file](https://cloud.githubusercontent.com/assets/631728/20680801/81216c38-b598-11e6-9b76-c4cc1f63dc0c.png)

**Comment by rui-castro** [comment]
@luis100 I will clarify the output.

About the format designation missing, format designation is composed of 2 attributes (formatDesignationName and formatDesignationVersion), if one of these is missing the formatDesignation is also missing.
I can make this more obvious changing the output message from **"Missing format designation"** to **"Missing format designation (name and/or version)"**. Let me know...

**Comment by rui-castro** [comment]
@luis100 new commit with improved output.

![screenshot from 2016-11-30 12 13 00](https://cloud.githubusercontent.com/assets/4218366/20751933/74c86522-b6f6-11e6-804c-fa17fc81becb.png)

Still waiting for answer in the format designation text.

**Comment by hsilva-keep** [comment]
@rui-castro Great great would be if you, instead of creating lines of text in the report details, create a ValidationReport, which then can be added in HTML version to the details, which will make the things look nicer. Take https://github.com/keeps/roda/blob/hsilva_dev/roda-core/roda-core/src/main/java/org/roda/core/plugins/plugins/base/AIPCorruptionRiskAssessmentPlugin.java#L197 & https://github.com/keeps/roda/blob/hsilva_dev/roda-core/roda-core/src/main/java/org/roda/core/plugins/plugins/base/AIPCorruptionRiskAssessmentPlugin.java#L206 as example.

**Comment by rui-castro** [comment]
@hsilva-keep it's a good idea, but for a future issue.

**Comment by luis100** [comment]
- [ ] Create new plugin parameter to verify if format designation version exists
- [ ] Separate check for format designation and format designation version

**Comment by rui-castro** [comment]
@luis100 new commit in PR #715 with the new parameter

**Comment by luis100** [comment]
- [ ] Ignore directories
Report failure and incidence are being create when selected file is a directory, it should be ignored.

**Comment by luis100** [comment]
Merged into lf_dev at 127f305e16bb6055b6eec53cb1c9d387d265883d, but missing the ignore directories.

**Comment by rui-castro** [comment]
@luis100 done in PR #736

**Comment by luis100** [comment]
Counters are not being calculated properly. Must count a folder as a success.

![formatnotcharacterizedcounters](https://cloud.githubusercontent.com/assets/631728/21095413/2940ba94-c053-11e6-9692-0e1faf550bf6.png)

**Comment by rui-castro** [comment]
Ok. I will fix it.

Em 12/12/2016 10:10, "Luis Faria" <notifications@github.com> escreveu:

Counters are not being calculated properly. Must count a folder as a
success.

[image: formatnotcharacterizedcounters]
<https://cloud.githubusercontent.com/assets/631728/21095413/2940ba94-c053-11e6-9692-0e1faf550bf6.png>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/keeps/roda/issues/642#issuecomment-266389733>, or mute
the thread

**Comment by rui-castro** [comment]
@luis100 done in PR #736

**Comment by rui-castro** [comment]
@luis100 ping

**Comment by luis100** [comment]
@rui-castro  pong

Checked and merged into lf_dev at c6fac937d3cde741640827e58b4da35b68f874c3

**Comment by rui-castro** [comment]
@luis100 :+1:
